google_vision_python/main.py

```diff
@@ -0,0 +1,153 @@
+import urllib.request
+import os
+import json
+from google.cloud import storage, vision
+import google.oauth2.credentials
+from google.cloud import pubsub_v1
+import constants
+import base64
+import io
+import time
+import logging
+
+BASE_TEMP_DIR = "/tmp/"
+
+os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = constants.google_vision_api_key_path
+
+def store_output_to_bucket(event, context):
+
+    pubsub_data_string = event['data']
+    logging.error(pubsub_data_string)
+    pubsub_message = base64.b64decode(pubsub_data_string).decode('utf-8')
+    data = json.loads(pubsub_message, strict=False)
+
+    uuid = None
+    variant_id = None
+    project_id = None
+    file_name = None
+    source_path = None
+    destination_path = None
+    bucket_access_token = None
+    bucket_name = None
+    topic_name = None
+
+    if data.get('metadata'):
+        uuid = data.get('metadata').get('uuid')
+        variant_id = data.get('metadata').get('variantId')
+        project_id = data.get('metadata').get('projectId')
+
+    if data.get('bucket'):
+        if data.get('bucket').get('source'):
+            file_name = data.get('bucket').get('source').get('file')
+            source_path = data.get('bucket').get('source').get('path')
+        if data.get('bucket').get('destination'):
+            destination_path = data.get('bucket').get('destination').get('path')
+            bucket_access_token = data.get('bucket').get('destination').get('accessToken')
+            bucket_name = data.get('bucket').get('destination').get('bucketName')
+   
+    if data.get('response'):
+        topic_name = data.get('response').get('topic')
+
+    bucket_credentials = google.oauth2.credentials.Credentials(bucket_access_token)
+
+    source_file_path = str(source_path) + '/' + str(file_name)
+
+    logging.error("Downloading file from bucket")
+    temp_file_path = download_from_bucket(bucket_name, file_name, project_id, bucket_credentials, source_file_path)
+
+    time.sleep(2)
+
```

Comment:
File will be downloaded in 2 sec?

Reply:
Can we go to the next step once the file is downloaded?

Reply:
> Can we go to the next step once the file is downloaded?

?

Reply:
> File will be downloaded in 2 sec?

Yes, for the most part 2 seconds is enough.

Reply:
We cannot assume all sizes of files will be downloaded in 2 seconds. Kindly make it async

Reply:
by sync do you mean to remove the sleep timer?

Reply:
Below code should run only after the file is downloaded..

  with io.open(temp_file_path, 'rb') as image:
        image_data = image.read()
    vision_api_json_response = run_vision_api_on_image(image_data)

Reply:
https://googleapis.dev/python/storage/latest/blobs.html